pkg/core/interop_neo.go

```diff
@@ -180,7 +180,13 @@ func (ic *interopContext) txGetUnspentCoins(v *vm.VM) error {
 	if err != nil {
 		return errors.New("no unspent coin state found")
 	}
-	v.Estack().PushVal(vm.NewInteropItem(ucs))
+	items := make([]vm.StackItem, len(ucs.States))
```

Comment:
You gonna get wrong length (or technically, wrong items being stored) because of filtering.